java-checks/src/main/java/org/sonar/java/checks/tests/NoTestInTestClassCheck.java

```diff
@@ -113,7 +113,11 @@ private static boolean isArchUnitTestClass(Symbol.TypeSymbol classSymbol) {
   }
 
   private void checkTestNGmembers(IdentifierTree className, Stream<Symbol> members) {
-    if (members.noneMatch(member -> member.isMethodSymbol() && member.isPublic() && !member.isStatic() && !"<init>".equals(member.name()))) {
+    if (members.noneMatch(member -> {
+      boolean annotatedWithTest = isTestFieldOrMethod(member);
+      boolean publicMethod = member.isMethodSymbol() && member.isPublic() && !member.isStatic() && !"<init>".equals(member.name());
+      return annotatedWithTest || publicMethod;
+    })) {
```

Comment:
Can we extract the second test into its own method and simplify the code?
```suggestion
    if (members.noneMatch(member -> isTestFieldOrMethod(member) || isPublicInstanceMethod(member)) {
```